runtime/major_gc.c

```diff
@@ -1409,12 +1420,16 @@ static intnat major_collection_slice(intnat howmuch,
     }
   }
 
+  if (mode == Slice_interruptible) {
+    *interrupted_budget_out = interrupted_budget;
+  }
+
   return budget;
```

Comment:
We need to think a bit about the return value of `major_collection_slice`: AFAICT it is never used by the caller. I think it could/should be `budget + interrupted_budget`, but really what do we want to use it for?
Maybe you should just hijack it to return `interrupted_budget`.

Reply:
You are right, the return value is no longer used (previously it had been used by the `Domain.Sync` blocking interface). 

If we want to hijack the return then I think it has to be `interrupted_budget` rather than `budget + interrupted_budget` because we need to differentiate:
- there is no marking/sweeping left to do so there is `budget` left over
vs
- you got interrupted to service a STW, so have work you should try to do later

Reply:
I've implemented `return interrupted_budget` for `major_collection_slice`. I also dropped (the unused) returns from `caml_opportunistic_major_collection_slice` and `caml_major_collection_slice`.

Thanks for the review and pushing for this. I think it has made the code better.